File: gpu_scraper.js
/**
 * RTX 5070 Ti 京东价格爬虫
 */

const https = require('https');

const BRANDS = [
    "七彩虹 5070 Ti",
    "华硕 5070 Ti",
    "微星 5070 Ti",
    "技嘉 5070 Ti",
    "影驰 5070 Ti",
    "索泰 5070 Ti"
];

function search(keyword) {
    return new Promise((resolve) => {
        const url = `https://search.jd.com/Search?keyword=${encodeURIComponent(keyword)}&enc=utf-8`;
        
        https.get(url, {
            headers: {
                'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
            }
        }, (res) => {
            let data = '';
            res.on('data', chunk => data += chunk);
            res.on('end', () => {
                console.log(`✓ ${keyword}: ${data.length} bytes`);
                resolve(data.length > 0);
            });
        }).on('error', (e) => {
            console.log(`✗ ${keyword}: ${e.message}`);
            resolve(false);
        });
    });
}

async function main() {
    console.log("=" .repeat(60));
    console.log("RTX 5070 Ti 京东价格爬虫");
    console.log("=" .repeat(60));
    
    for (const brand of BRANDS) {
        await search(brand);
    }
    
    console.log("\n提示：京东有反爬，完整数据需要 Selenium 或 API");
}

main();

File: gpu_scraper.py
#!/usr/bin/env python3
"""
RTX 5070 Ti 京东价格爬虫
抓取多个品牌的 5070 Ti 显卡价格和基本信息
"""

import requests
import json
from datetime import datetime

# 京东搜索 API（模拟）
SEARCH_URL = "https://search.jd.com/Search"

# 常见 5070 Ti 品牌系列
BRANDS = [
    "七彩虹 5070 Ti",
    "华硕 5070 Ti", 
    "微星 5070 Ti",
    "技嘉 5070 Ti",
    "影驰 5070 Ti",
    "索泰 5070 Ti",
    "映众 5070 Ti",
    "铭瑄 5070 Ti"
]

def search_jd(keyword):
    """搜索京东商品"""
    params = {
        "keyword": keyword,
        "enc": "utf-8",
        "qrst": "1",
        "rt": "1",
        "stop": "1",
        "vt": "2",
        "wq": keyword,
        "stock": "1",
    }
    
    headers = {
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36",
        "Referer": "https://www.jd.com/"
    }
    
    try:
        resp = requests.get(SEARCH_URL, params=params, headers=headers, timeout=10)
        return resp.text
    except Exception as e:
        print(f"搜索失败 {keyword}: {e}")
        return None

def parse_products(html, brand):
    """简单解析商品列表"""
    products = []
    if not html:
        return products
    
    # 这里需要更复杂的解析，先返回占位数据
    return products

def main():
    print("=" * 60)
    print("RTX 5070 Ti 京东价格爬虫")
    print(f"时间：{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    print("=" * 60)
    
    results = []
    
    for brand in BRANDS:
        print(f"\n搜索：{brand}")
        html = search_jd(brand)
        if html:
            print(f"  ✓ 获取到数据 ({len(html)} bytes)")
            # 后续解析逻辑
        else:
            print(f"  ✗ 获取失败")
    
    print("\n" + "=" * 60)
    print("爬虫完成")
    
    # 由于京东反爬，可能需要 Selenium
    print("\n提示：京东有反爬机制，如需完整数据建议使用 Selenium")

if __name__ == "__main__":
    main()
